src/ESLint.Formatter/sarif-with-rules.js

```diff
@@ -0,0 +1,145 @@
+
+            // Create a new entry in the files dictionary.
+            sarifFiles[result.filePath] = {
+                fileLocation: {
+                    uri: result.filePath
+                }
+            };
+        }
+
+        const messages = result.messages;
+
+        messages.forEach(message => {
+
+            const sarifResult = {
+                level: getResultLevel(message),
+                message: {
+                    text: message.message
+                },
+                locations: [
+                    {
+                        physicalLocation: {
+                            fileLocation: {
+                                uri: result.filePath
+                            }
+                        }
+                    }
+                ]
+            };
+
+            if (message.ruleId) {
+                sarifResult.ruleId = message.ruleId;
+
+                if (typeof sarifRules[message.ruleId] === "undefined") {
+                    const rule = rules.get(message.ruleId);
+
+                    // An unknown ruleId will return null. This check prevents unit test failure.
+                    if (rule) {
+
+                        // Create a new entry in the rules dictionary.
+                        sarifRules[message.ruleId] = {
+                            id: message.ruleId,
+                            shortDescription: {
+                                text: rule.meta.docs.description
+                            },
+                            helpUri: rule.meta.docs.url,
+                            properties: {
+                                category: rule.meta.docs.category
+                            }
+                        };
+                    }
+                }
+            }
+
+            if (message.line > 0 || message.column > 0) {
+                sarifResult.locations[0].physicalLocation.region = {
+                    startLine: message.line,
+                    startColumn: message.column
+                };
+            }
+
+            if (message.source) {
+
+                // Create an empty region if we don't already have one from the line / column block above.
+                sarifResult.locations[0].physicalLocation.region = sarifResult.locations[0].physicalLocation.region || {};
+                sarifResult.locations[0].physicalLocation.region.snippet = {
+                    text: message.source
+                };
+            }
+
+            sarifResults.push(sarifResult);
+
+        });
+
+    });
+
+    //------------------------------------------------------------------------------
+    // We use separate arrays and dictionaries to avoid creating empty properties in the SARIF log.
+    // Here we set those properties accordingly.
+    //------------------------------------------------------------------------------
+
+    if (Object.keys(sarifFiles).length > 0) {
+        sarifLog.runs[0].files = sarifFiles;
+    }
+
+    if (sarifResults.length > 0) {
```

Comment:
Because this is an array, not a dictionary.

---
In reply to: [249142453](https://github.com/Microsoft/sarif-sdk/pull/1222#discussion_r249142453) [](ancestors = 249142453)